### PR TITLE
Add endpoints to get campaign activity

### PR DESCRIPTION
**Pull request by iamacook**
## Summary

The designs of Safe{Pass} require points of the current campaign/period be displayed. As such, a new endpoint is [being implemented on the Locking Service](https://github.com/safe-global/safe-locking-service/issues/131) to return this information.

This integrates the aformentioned into the Gateway under `GET /v1/community/campaigns/:resourceId/activities`, accepting a `holder` query parameter for specific addresses.

## Changes

- Add new Locking Service endpoint to `ILockingApi`
- Add `CampaignPointsSchema`/`CampaignPointsPageSchema` with relevant entities and builders
- Add `ICommunityRepository['getCampaignPointsForAddress']`, validating against `CampaignPointsPageSchema`
- Add `GET /v1/community/campaigns/:resourceId/activities` to `CommunityController`/`CommunityService`
- Add/update tests accordingly

### CODE REVIEW

**Comment by iamacook** [comment]
I'll leave this as a draft [until the endpoint/API has been finalised](https://github.com/safe-global/safe-locking-service/issues/131#issuecomment-2158583089).

File: src/routes/community/community.controller.ts [hunk 1]
```diff
@@ -44,6 +45,22 @@ export class CommunityController {
     return this.communityService.getCampaignById(resourceId);
   }
 
+  @Get('/campaigns/:resourceId/points/:safeAddress')
```
Comment (iamacook) [inline src/routes/community/community.controller.ts:48]: After discussing this with Core, this is quite restrictive. The entity returned is actually a flattened `Period` and `Activity` so we should rethink this.

They are opting for `/api/v1/campaigns/{resourceId}/addresses/{address}/periods`, for example.

File: src/routes/community/community.controller.ts [hunk 2]
```diff
@@ -44,6 +45,22 @@ export class CommunityController {
     return this.communityService.getCampaignById(resourceId);
   }
 
+  @Get('/campaigns/:resourceId/activities')
+  async getCampaignActivities(
+    @Param('resourceId') resourceId: string,
+    @RouteUrlDecorator() routeUrl: URL,
+    @PaginationDataDecorator() paginationData: PaginationData,
+    @Query('holder', new ValidationPipe(AddressSchema.optional()))
```
Comment (hectorgomezv) [inline src/routes/community/community.controller.ts:53]: A file intended to validate the `holder` (`src/routes/community/entities/campaign-activity-holder.dto.entity.ts`) was also included by mistake I think.

File: src/domain/community/community.repository.interface.ts [hunk 3]
```diff
@@ -14,6 +15,11 @@ export interface ICommunityRepository {
     offset?: number;
   }): Promise<Page<Campaign>>;
 
+  getCampaignActivities(args: {
+    resourceId: string;
+    holder?: `0x${string}`;
```
Comment (hectorgomezv) [inline src/domain/community/community.repository.interface.ts:20]:
```
  limit?: number;
  offset?: number;
```

This is missing in the function interface, right?

File: src/routes/community/community.controller.spec.ts [hunk 4]
```diff
@@ -279,6 +283,155 @@ describe('Community (Unit)', () => {
...
+        .get(
+          `/v1/community/campaigns/${campaign.resourceId}/activities?holder=${holder}`,
+        )
+        .expect(200)
+        .expect({
+          count: 1,
+          next: null,
+          previous: null,
+          results: [campaignActivityToJson(campaignActivity)],
+        });
```
Comment (hectorgomezv) [inline src/routes/community/community.controller.spec.ts:347]: Nit: could we include a test verifying the `networkService` GET call included the `holder`, `limit` and `offset` parameter?

**Review by hectorgomezv — CHANGES_REQUESTED** [review]
Some nitpick comments only!

File: src/routes/community/community.controller.ts (hunk 2 above)
Comment (iamacook) [reply src/routes/community/community.controller.ts:53]: Removed in 4d67562054be2d64dd9b8a6648fede25cf685324.

File: src/domain/community/community.repository.interface.ts (hunk 3 above)
Comment (iamacook) [reply src/domain/community/community.repository.interface.ts:20]: It was! Added in 4d67562054be2d64dd9b8a6648fede25cf685324.

File: src/routes/community/community.controller.spec.ts (hunk 4 above)
Comment (iamacook) [reply src/routes/community/community.controller.spec.ts:347]: Added in 724dc1318d1bd6cb4b5ca3fc141930b0f4edbbe5.

**Review by hectorgomezv — APPROVED** [review]

**Pull request merged**